test/e2e/create_test.go

```diff
@@ -74,13 +75,15 @@ func testWindowsNodeCreation(t *testing.T) {
 		err = testCtx.waitForWindowsNodes(gc.numberOfNodes, true, false, false)
 		assert.NoError(t, err, "Windows node creation failed")
 
-		t.Run("Changing the private key ensures all Windows nodes use the new private key", func(t *testing.T) {
-			// Replace private key and check that new Machines are created using the new private key
-			err = testCtx.createPrivateKeySecret(true)
-			require.NoError(t, err, "error replacing private key secret")
-			err = testCtx.waitForWindowsNodes(gc.numberOfNodes, true, false, false)
-			assert.NoError(t, err, "error waiting for Windows nodes configured with newly created private key")
-		})
+		if testCtx.CloudProvider.GetType() != config.VSpherePlatformType {
+			t.Run("Changing the private key ensures all Windows nodes use the new private key", func(t *testing.T) {
```

Comment:
This is actually the one test that should work. The lines that are going to cause failure are the ones where false is used in `testCtx.createPrivateKeySecret(false)`. When false is passed in a randomly generated key is used.

Reply:
BTW my thought process behind using the known private key for the final test was so that if there were issues with the VM in the rest of the test suite they could be more easily debugged by a person, as they wouldnt have to decode the secret to get the private key to use.

Reply:
As discussed I have flipped the ordering of known vs unknown private keys and only doing the key flipping tests on AWS.